Makefile

```diff
@@ -235,6 +240,14 @@ python/format:
 	black $(PYTHON_FILES)
 	@ echo "[$@]: Successfully formatted Python files!"
 
+## Formats Go files
+golang/%: GOLANG_FILES ?= $(shell git ls-files --cached --others --exclude-standard '*.go')
+golang/format: | guard/program/goimports
+golang/format:
+	@ echo "[$@]: Formatting Golang files..."
+	echo $(GOLANG_FILES) | xargs -r goimports -l -w | grep . && exit 1 || exit 0
```

Comment:
It looks like `-w` option writes the changes back to source file. Making changes is fine for the `golang/format` target, but then we _wouldn't_ exit non-zero on changes.

What do you think of incorporating this "changes detected" logic in a `golang/lint` target... Using `-d` instead of `-w` to just print the diff. And inverting the `grep` with `-v` to avoid needing to control the exit codes...

```
golang/lint: | guard/program/goimports
	@ echo "[$@]: Linting Golang files..."
	echo $(GOLANG_FILES) | xargs -r goimports -l -d | grep -v .
	@ echo "[$@]: Successfully linted Golang files!"
```

Reply:
The target for `black` modifies the files and it was the `python/lint` target that warned of changes but didn't automatically make any formatting changes.  I was trying to be consistent in that respect.

I don't really care ... there isn't a consistent philosophy with this Makefile and I just have to guess what you're going to accept or not accept.  I have yet to guess correctly.

Reply:
It is most certainly consistent. `%/lint` makes no changes. These targets are called in CI and exit error on changes. `%/format` makes changes. These targets are called by users to fix styling. I'm asking for the same design here for the `golang/lint` and `golang/format` targets.

Reply:
The `golang/lint` target is not something we were going to add to the Makefile.  There are several tools that could do lint golang files, but the ones I'm interested want to work on packages versus files, or require reasonable go.mod and go.sum file.  The idea was to add a `golang/lint` on a as-needed basis to project Makefiles.

`goimports` is a formatting tool.  It does the work of `gofmt` in addition to checking the list of imports to see if they're used.

Reply:
Ok, but as written this target is actually linting, with the error exit when it makes changes. It is valid to have a linter that is purely about formatting style. When in linting mode, it would not make changes, it would just error exit if it detected changes it wants to make. When in formatting mode, it would exit with success if it was able to format the files. This is exactly what we do with black today, when it runs in the python/lint and python/format targets. I don't really see any conflict with doing the same thing here with golang files.

This target needs to decide what it wants to be. Is it a linter or a formatter? If it remains a formatter, then remove it from the lint target, and let it exit with success when it succeeds at making changes. If it will be a linter, then change the name, run the tool without making changes, and have it error exit if it detect changes. Or do both in two separate targets.